IoTDeviceSimulator/web-client/src/BehaviourComponent.js

```diff
@@ -43,14 +43,32 @@ export function SineComponent(props) {
     }
     return (
         <>
-            <div className="behaviour-value">
-                <LabeledInput className="labels" type='number' label='Mean' name='mean' value={deviceMean} onChange={changeMean} disabled={props.signalArray !== "" ? true : false} />
-                <LabeledInput className="labels" type='number' label='Amplitude' name='amplitude' value={deviceAmplitude} onChange={changeAmplitude} disabled={props.signalArray !== "" ? true : false} />
-                <LabeledInput className="labels" type='number' label='Period(ms)' name='wave_period' value={deviceWavePeriod} onChange={changeWave_period} disabled={props.signalArray !== "" ? true : false} />
-            </div >
-            <ChartComponent labelsArray={arr} dataArray={ar} chartName="Component Signal" />
+            {props.isAdmin === true ?
+            <>
+                <div className="behaviour-value">
+                    <LabeledInput className="labels" type='number' label='Mean' name='mean' value={deviceMean} onChange={changeMean} disabled={props.signalArray !== "" ? true : false} />
+                    <LabeledInput className="labels" type='number' label='Amplitude' name='amplitude' value={deviceAmplitude} onChange={changeAmplitude} disabled={props.signalArray !== "" ? true : false} />
+                    <LabeledInput className="labels" type='number' label='Period(ms)' name='wave_period' value={deviceWavePeriod} onChange={changeWave_period} disabled={props.signalArray !== "" ? true : false} />
+                </div >
+            
+                <ChartComponent labelsArray={arr} dataArray={ar} chartName="Component Signal" />
```

Comment:
take this out save in JS variable and inside return that variable. Do similar things for other.

Reply:
Address this comment in a subsequent PR.